client-skeleton-common/src/main/java/eu/arrowhead/client/skeleton/common/ArrowheadService.java

```diff
@@ -0,0 +1,395 @@
+	 * @throws BadPayloadException when the payload couldn't be validated by Orchestrator Core System 
+	 * @throws InvalidParameterException when the payload content couldn't be validated by Orchestrator Core System
+	 * @throws ArrowheadException when internal server error happened at one of the core system involved in orchestration process 
+	 * @throws UnavailableServerException when one of the core system involved in orchestration process is not available 
+	 */
+	public ResponseEntity<OrchestrationResponseDTO> proceedOrchestration(final OrchestrationFormRequestDTO request) {
+		final CoreServiceUri uri = getCoreServiceUri(CoreSystemService.ORCHESTRATION_SERVICE);
+		if (uri == null) {
+			logger.debug("Orchestration couldn't be proceeded due to the following reason: " +  CoreSystemService.ORCHESTRATION_SERVICE.name() + " not known by Arrowhead Context");
+			return null;
+		}
+		
+		return httpService.sendRequest(Utilities.createURI(getUriScheme(), uri.getAddress(), uri.getPort(), uri.getPath()), HttpMethod.POST, OrchestrationResponseDTO.class, request);
+	}
+	
+	//-------------------------------------------------------------------------------------------------
+	/**
+	 * Sends a http(s) request with the specified service reachability details.
+	 * 
+	 * @param httpMethod HttpMethod enum which represents the method how the service is available.
+	 * @param address String value which represents the host where the service is available.
+	 * @param port int value which represents the port where the service is available
+	 * @param serviceUri String value which represents the URI where the service is available.
+	 * @param interfaceName String value which represents the name of the interface used for the communication. Usable interfaces could be received in orchestration response.
+	 * @param token (nullable) String value which represents the token for being authorized at the provider side if necessary. Token could be received in orchestration response per interface type.  
+	 * @param payload (nullable) Object type which represents the required payload of the http(s) request if any necessary.
+	 * @param queryParams (nullable) String... variable arguments which represent the additional key-value http(s) query parameters if any necessary. E.g.: "k1", "v1", "k2", "v2".  
+	 * @return the response received from the provider 
+	 * 
+	 * @throws InvalidParameterException when service URL can't be assembled.
+	 * @throws AuthException when ssl context or access control related issue happened.
+	 * @throws ArrowheadException when the communication is managed via Gateway Core System and internal server error happened.
+	 * @throws UnavailableServerException when the specified server is not available.
+	 */
+	public ResponseEntity<Object> consumeServiceHTTP(final HttpMethod httpMethod, final String address, final int port, final String serviceUri, final String interfaceName, final String token,
```

Comment:
make this method generic. Return value should be T,  and you need to add a new parameter: final Class<T> responseType